src/test/java/org/openmrs/addonindex/rest/CoreVersionsControllerIT.java

```diff
@@ -0,0 +1,91 @@
+package org.openmrs.addonindex.rest;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.openmrs.addonindex.domain.AddOnInfoAndVersions;
+import org.openmrs.addonindex.domain.AddOnVersion;
+import org.openmrs.addonindex.domain.artifactory.VersionList;
+import org.openmrs.addonindex.rest.artifactory.CoreVersionsController;
+import org.openmrs.addonindex.service.artifactory.VersionsService;
+import org.openmrs.addonindex.util.Version;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.embedded.LocalServerPort;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.when;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class CoreVersionsControllerIT {
+    @LocalServerPort
+    private int port;
+
+    @Autowired
+    private TestRestTemplate testRestTemplate;
+
+    @MockBean
+    private VersionsService versionsService;
+
+    @Autowired
+    private CoreVersionsController controller;
+
+    private VersionList versionList;
+
+    @Before
+    public void setUp() throws Exception {
+        List<String> versions = new ArrayList<>();
+        versions.add("1.6.3");
+        versions.add("1.6.4");
+        VersionList versionList;
+        versionList = new VersionList(versions);
+        doReturn(versionList).when(versionsService).getVersions();
+
+    }
+    @Test
+    public void getCoreversions() throws Exception {
+        ResponseEntity<String> entity = testRestTemplate.getForEntity("http://localhost:" + port + "/api/v1/coreversions",
+                String.class);
+
+        assertThat(entity.getStatusCode(), is(HttpStatus.OK));
+        assertThat(entity.getBody(),is("[\"1.6.3\",\"1.6.4\"]"));
+    }
+
+    @Test
+    public void getLatestSupportedVersionTest() throws Exception{
```

Comment:
From the perspective of testing strategy, this test is bad.

The value of having a (slow, expensive) integration test on a controller is to do an end-to-end test of the system. But in this case you're mocking VersionsService so we don't actually get an end-to-end test out of this. Instead we get a slow test that's retesting the same behavior that should be tested in the unit test for AddOnInfoAndVersions.

Instead of mocking the behavior of `versionsService.getMatchedVersion("2.1.0","testing-module")` you should have a @MockBean Index and mock the `index.getByUid("testing-module")`.

Also, a side-comment is that you're setting up _more_ test data in this test as compared to AddOnInfoAndVersionsTest. Instead you should be testing more of the business logic in that (faster, cheaper) unit test, and in this integration test you just test a single happy path.
(I'm not saying to remove one of the versions from this test, just that you should add more versions in the other test.)